django-backend/scripts/generate_transaction_record_fixture.py

```diff
@@ -0,0 +1,269 @@
+
+
+def get_date_entry(date_field, date):
+    if (len(date_field) > 0):
+        return f''',\n{" "*12}"{date_field}": "{date}"'''
+    return ""
+
+
+def get_report_id(args, test_case):
+    if "within_dates" in test_case:
+        return args.report_a_id
+    else:
+        return args.report_b_id
+
+
+def random_hex(length=1):
+    hex_chars = "0123456789abcdef"
+
+    out_string = ""
+    while len(out_string) < length:
+        out_string += choice(hex_chars)
+
+    return out_string
+
+
+def get_id(form_type, record_number, schedule=""):
+    hex_chars = "0123456789abcdef"
+
+    schedule_str = ""
+    if(schedule[0].lower() in hex_chars):
+        schedule_str += schedule.lower()
+    else:
+        schedule_str += hex(ord(schedule.lower()))[2:]
+
+    line_number = form_type.split(schedule)[1].strip('/')
+    schedule_str += "0" + str(line_number)
+    schedule_str = schedule_str.ljust(8, "0")
+
+    record_str = str(record_number + 1).rjust(12, "0")
+
+    return f"{schedule_str}-{random_hex(4)}-{random_hex(4)}-{random_hex(4)}-{record_str}"
+
+
+def get_records(form_type, args):
+    records = []
+    test_cases = [
+        "within_dates",
+        "within_dates",
+        "within_year",
+        "outside_dates",
+        "within_dates_but_memo"
+    ]
+    for record_number in range(len(test_cases)):
+        test_case = test_cases[record_number]
+
+        comment = get_comment(form_type, test_case)
+
+        schedule = get_schedule(form_type)
+        schedule_model = f"schedule{schedule.lower()}"
+
+        schedule_id = get_id(form_type, record_number, schedule)
+        transaction_id = get_id(form_type, record_number, schedule)
+
+        amount_field = get_amount_field(schedule)
+        amount = choice(range(20, 150))
+
+        date_field = get_date_field(schedule)
+        date = get_date(args, test_case)
+        date_entry = get_date_entry(date_field, date)
+
+        report_id = get_report_id(args, test_case)
+
+        memo_code = str(
+            test_case == "within_dates_but_memo"
+        ).lower()  # JSON uses lowercase true/false
+
+        schedule_record = f"""    {{
+        "comment": "{comment}",
+        "model": "transactions.{schedule_model}",
+        "fields": {{
+            "id": "{schedule_id}",
+            "{amount_field}": {amount}{date_entry}
+        }}\n    }}"""
+
+        transaction_record = f"""    {{
+        "comment": "{comment}",
+        "model": "transactions.transaction",
+        "fields": {{
+            "id": "{transaction_id}",
+            "schedule_{schedule.lower()}_id": "{schedule_id}",
+            "committee_account_id": "{args.committee_account_id}",
+            "report_id": "{report_id}",
+            "_form_type": "{form_type}",
+            "memo_code": {memo_code},
+            "created": "2022-02-09T00:00:00.000Z",
+            "updated": "2022-02-09T00:00:00.000Z",
+            "transaction_type_identifier": "Transaction Type Identifier"
```

Comment:
noticing that this got carried through to the test data.  I guess the summary logic only acts over form_type so as long as the test data is only used there we're ok

Reply:
I've added a flag that lets you optionally override the TTI, though it still defaults to a meaningless "Transaction Type Identifier"